components/onboarding/internal/services/query/get-all-metadata-account-type.go

```
@@ -9,31 +9,41 @@ import (
 
 	metadata, err := uc.MetadataRepo.FindList(ctx, reflect.TypeOf(mmodel.AccountType{}).Name(), filter)
-	if err != nil || metadata == nil {
+	if err != nil {
+		libOpentelemetry.HandleSpanError(span, "Failed to get metadata on repo", err)
+
+		logger.Log(ctx, libLog.LevelError, "Error getting account type metadata on repo")
+
+		return nil, libHTTP.CursorPagination{}, err
+	}
+
+	if metadata == nil || len(metadata) == 0 {
```

golangci-lint (2.11.3)

[error] S1009: should omit nil check; len() for nil slices is defined as zero

(staticcheck)